### PR TITLE
impl(bigtable): support bigtable cookie in `SampleRows`

### PR DESCRIPTION
**Pull request by dbolduc**
Part of the work for #13447 

More of the same...

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13477)

### CODE REVIEW

File: google/cloud/bigtable/internal/data_connection_impl_test.cc [hunk 1]
```diff
@@ -1474,6 +1471,42 @@ TEST_F(DataConnectionTest, SampleRowsPermanentError) {
...
+          [this](auto context, auto const&, v2::SampleRowKeysRequest const&) {
+            auto headers = metadata_fixture_.GetMetadata(*context);
+            EXPECT_THAT(headers,
+                        Contains(Pair("x-goog-cbt-cookie-routing", "routing")));
+            auto stream = std::make_unique<MockSampleRowKeysStream>();
+            EXPECT_CALL(*stream, Read).WillOnce(Return(PermanentError()));
+            return stream;
+          });
+
+  auto mock_b = std::make_unique<MockBackoffPolicy>();
```
Comment (alevenberg) [inline google/cloud/bigtable/internal/data_connection_impl_test.cc:1495]: rename policy_mock?

File: google/cloud/bigtable/internal/data_connection_impl_test.cc [hunk 2]
```diff
@@ -1474,6 +1471,42 @@ TEST_F(DataConnectionTest, SampleRowsPermanentError) {
   EXPECT_THAT(samples, StatusIs(StatusCode::kPermissionDenied));
 }
 
+TEST_F(DataConnectionTest, SampleRowsBigtableCookie) {
+  auto mock = std::make_shared<MockBigtableStub>();
+  EXPECT_CALL(*mock, SampleRowKeys)
+      .WillOnce(
```
Comment (alevenberg) [inline google/cloud/bigtable/internal/data_connection_impl_test.cc:1477]: Why does this need to return twice?

Comment (dbolduc) [reply google/cloud/bigtable/internal/data_connection_impl_test.cc:1477]: In the first request, the server gives us a cookie.

In the second request, we give that cookie back to the server.

File: google/cloud/bigtable/internal/data_connection_impl_test.cc (hunk 1 above)
Comment (dbolduc) [reply google/cloud/bigtable/internal/data_connection_impl_test.cc:1495]: No thanks. The name is consistent with other tests in the file.

File: google/cloud/bigtable/internal/data_connection_impl_test.cc (hunk 2 above)
Comment (dbolduc) [reply google/cloud/bigtable/internal/data_connection_impl_test.cc:1486]: Added comments for this, and all other `*BigtableCookie` tests.

**Review by alevenberg — APPROVED** [review]

**Pull request merged**